pkg/sql/catalog/replication/reader_catalog_test.go

```diff
@@ -176,6 +176,14 @@ func TestReaderCatalog(t *testing.T) {
 	compareConn("SELECT * FROM t1 ORDER BY n")
 	compareConn("SELECT * FROM v1 ORDER BY 1")
 	compareConn("SELECT * FROM t2 ORDER BY j")
+
+	// Validate that schema changes are blocked.
```

Comment:
nice! could you add one test case related to modifying an existing descriptor in the reader tenant?